### PR TITLE
Improve font loading

### PR DESCRIPTION
**Pull request by coreysprague**
Reworked the mechanism for loading fonts, and improved the documentation.

Calling `await loadAppFonts()` will now automatically detect and include all fonts referenced in a package's `pubspec.yaml` as well as any fonts included in packages you depend on.

- Removed MaterialIcons from this package. It turns out they can be included automatically by ensuring that `use-material-design:  true` is in the `pubspec.yaml`.

- Roboto is now automatically loaded via the GoldenToolkit.

- Provided a mechanism for Cupertino fonts to be supplied by the consumer if needed.

- Adjusted the guidance and implementation of when to call `loadAppFonts()` by using `flutter_test_config.dart`

### CODE REVIEW

File: packages/golden_toolkit/lib/src/font_loader.dart [hunk 1]
```diff
@@ -1,57 +1,58 @@
...
 import 'package:flutter/services.dart';
-import 'package:meta/meta.dart';
+import 'package:flutter_test/flutter_test.dart';
 
 ///By default, flutter test only uses a single "test" font called Ahem.
 ///
 ///This font is designed to show black spaces for every character and icon. This obviously makes goldens much less valuable.
 ///
 ///To make the goldens more useful, we have a utility to dynamically inject additional fonts into the flutter test engine so that we can get more human viewable output.
 /// Path to your folder with fonts [from] in required
```
Comment (mZadorskii) [inline packages/golden_toolkit/lib/src/font_loader.dart]:
```suggestion
/// Paths to your folders with fonts [sourceDirectories]
```

Comment (mZadorskii) [reply packages/golden_toolkit/lib/src/font_loader.dart]: wait is it used?

Comment (coreysprague) [reply packages/golden_toolkit/lib/src/font_loader.dart]: nope, that was a miss!

Comment (coreysprague) [reply packages/golden_toolkit/lib/src/font_loader.dart]: fixed

File: packages/golden_toolkit/lib/src/font_loader.dart [hunk 2]
```diff
@@ -1,57 +1,58 @@
+import 'dart:convert';
```
Comment (mZadorskii) [inline packages/golden_toolkit/lib/src/font_loader.dart]: minor: should be below Copyright

File: packages/golden_toolkit/test/flutter_test_config.dart [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import 'dart:async';
```
Comment (mZadorskii) [inline packages/golden_toolkit/test/flutter_test_config.dart]:
```suggestion
/// ***************************************************
/// Copyright 2019-2020 eBay Inc.
///
/// Use of this source code is governed by a BSD-style
/// license that can be found in the LICENSE file or at
/// https://opensource.org/licenses/BSD-3-Clause
/// ***************************************************

import 'dart:async';
```

File: packages/golden_toolkit/test/font_loading_test.dart [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import 'dart:io';
```
Comment (mZadorskii) [inline packages/golden_toolkit/test/font_loading_test.dart]:
```suggestion
/// ***************************************************
/// Copyright 2019-2020 eBay Inc.
///
/// Use of this source code is governed by a BSD-style
/// license that can be found in the LICENSE file or at
/// https://opensource.org/licenses/BSD-3-Clause
/// ***************************************************

import 'dart:io';
```

File: packages/golden_toolkit/test/golden_builder_test.dart [hunk 5]
```diff
@@ -8,18 +8,13 @@
...
 import 'package:flutter/widgets.dart';
 import 'package:flutter_test/flutter_test.dart';
 
 import 'sample_weather_widget.dart';
 
 Future<void> main() async {
-  /// Note: In order to see fonts and icons on goldens,
-  /// you need to preload all the fonts with this function in all test files
-  await loadAppFonts(from: 'fonts');
-
```
Comment (mZadorskii) [inline packages/golden_toolkit/test/golden_builder_test.dart]: 👍

**Pull request merged**